Mathlib/Geometry/Manifold/Orientable.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rida Hamadani
+-/
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Orientable Manifolds
+
+This module defines orientable manifolds.
+
+## Main Definitions
+
+- `OrientationPreserving` : a map is orientation-preserving on a given set if the determinant of
+  its Jacobian is strictly positive on that set.
+- `orientationPreservingGroupoid` : the groupoid of partial homeos of `H` which are
+  orientation-preserving.
+- `OrientableManifold`: a type class saying that the charted space `M`, modelled on the space `H`,
+  admis an orientation.
+
+## TODO
+
+- Define orientable `0`-manifolds.
+
+-/
+
+variable {H : Type*} [NormedAddCommGroup H] [NormedSpace ℝ H]
+
+section OrientationPreserving
+
+/--
+A map is orientation-preserving on a given set if it is differentiable and the determinant of its
+Jacobian is strictly positive on that set.
+-/
+def OrientationPreserving (f : H → H) (s : Set H) : Prop :=
+  ∀ x ∈ s, 0 < (fderiv ℝ f x).det
+
+/--
+A map is orientation-reversing on a given set if it is differentiable and the determinant of its
+Jacobian is strictly negative on that set.
+-/
+def OrientationReversing (f : H → H) (s : Set H) : Prop :=
+  ∀ x ∈ s, (fderiv ℝ f x).det < 0
+
+lemma OrientationPreserving.DifferentiableAt [Nontrivial H] [FiniteDimensional ℝ H] {f : H → H}
```

Comment:
Three small comments: all of them also apply to the other lemmas below.
- I think you don't need the `Nontrivial H` assumption: if `H` is zero-dimensional, `f` is always differentiable. I think mathlib has this already; ping me if you get stuck finding it!
- By default, I would put the `x` "left of the colon", i.e. write "(x : s) : DifferentiableAt \R f x" instead. Do you have a reason to do this differently?
- In terms of naming, I'd write "OrientationPreserving.differentiableAt".

Reply:
Thank you for pointing these out! Regarding point two: instead of writing `(x : s)`, I decided to look at similar lemmas in mathlib and imitate how it is done there (e.g. `DifferentiableWithinAt.differentiableAt`).